src/banners/banners.service.ts

```diff
@@ -10,23 +18,123 @@ export class BannersService {
     @InjectRepository(BannersRepository)
     private bannersRepository: BannersRepository,
   ) {}
-  create(createBannerDto: CreateBannerDto) {
-    return this.bannersRepository.create({});
+
+  async create(icon, createBannerDto: CreateBannerDto): Promise<Banner> {
+    // return icon;
+    if (!icon?.filename) {
+      throw new BadRequestException();
+    }
+
+    createBannerDto.text = 'search';
+
+    if (isURL(createBannerDto.href)) {
+      createBannerDto.text = 'url';
+    }
+
+    // create path point to images resource
+    const path = `/banners/images/${icon?.filename}`;
+    createBannerDto.icon = path;
+
+    // save and assign a result saved
+    const saved = await this.bannersRepository.save(createBannerDto);
+
+    // if saved fail should be throw new error
+    if (!saved) {
+      throw new BadRequestException(`Can not save object`);
+    }
+
+    return saved;
   }
 
-  async findAll() {
-    return '';
+  async findAll(): Promise<Banner[]> {
+    const banners = await this.bannersRepository.find({ withDeleted: true });
+    return banners;
   }
 
-  findOne(id: number) {
-    return `This action returns a #${id} banner`;
+  async findOneBanner(id: string): Promise<Banner> {
+    // find one banner by id and with deleted
+    const banner = await this.bannersRepository.findOne({
+      where: {
+        id,
+      },
+      withDeleted: true,
+    });
```

Comment:
you want to support finding a soft-deleted record. Right?
I think you should also provide an options param to this method for purpose of finding record with/without soft-deleted.